### ISSUE TITLE
Fails to sync if changing state and note content both sides

### ISSUE THREAD

**Issue by ericzolf** [spacecowboy/NotePad#365]
If I change the "completed" state on my Android using this app, and change the content itself directly in Google Notes/Calendar and then try to sync, I manage to make Notes to bring the dreaded "failed to Sync" message. Deinstalling the app and reinstalling it makes the problem disappear, but of course local changes are lost.

To be honest, I'm not too sure in which order I need to carry the steps to make the sync issue appear, but it's repeatable and I got once rid of it by manually bringing both sides of the synced note back to the same state and content.

I see also as an issue that Notes doesn't bring more information about the sync issue it encountered, some more explicit error message would be nice and would allow us, users, to give you developers more explicit information.

**Comment by spacecowboy** [comment]
Thanks for your help in trying to solve this.

But following that recipe myself, I can't get it to fail :(

Can you still reproduce it after reinstalling the app?

Also, what version do you have installed?

**Comment by himanshuraja** [comment]
can I work on this?

**Comment by ericzolf** [comment]
@spacecowboy - sorry for the delay in answering.

I have regularly (not often but regularly) the problem and the only solution to fix it I found till own is to reinstall the app, and it appears again after a while, and this makes sure that I have the latest version from the Market.

I have the same effect with two different accounts, one for work, one private, which I use on two different smartphones and, the private one, a tablet.
Work: Samsung S6
Private: Sony Xperia Z1 compact and Google Nexus7 2012 (the last one with CyanogenMod).
Meaning a rather complete mix of different systems and versions (shouldn't be the issue).

Perhaps one more piece of information: I'm using quite extensively the Google Tasks Sync plug-in for Thunderbird in version 0.5.3 and seldomly the WebUI of Google Calendar, so that's perhaps important (the fact that Notes works again after a reinstall tells me still that the error is on its side, but who knows with IT).

@himanshuraja - any help is appreciated from my side.

**Comment by spacecowboy** [comment]
@himanshuraja feel free

@ericzolf thanks for the additional info. The problem is entirely the app's fault, but I'll see if the Thunderbird plugin does something special which can create the situation. Hopefully I can recreate it.

**Comment by ericzolf** [comment]
Hi,

I haven't seen anything happening and the sync issue is still there, I just got it and it's really annoying. Especially as it's again a bit different: in the past, it happened on first use (meaning, I open the app, it fails, I uninstall the app, re-install, sync is OK again). Now, I opened the app, synced OK, did some changes, tried to sync and it failed, meaning that I lost my changes.

Really, I think that the first thing to do is to make sure that the app spits more information about the actual issue so that a user (me!) can provide it back to such a bug.

As a side remark, clearing the cache isn't sufficient, clearing all data also solves the issue (and force stop).

I would really appreciate some progress on this issue. As said, I'm really to do two steps with first a version providing more information, which I'd attach to this bug, and then (hopefully) the real fix.

Thanks, Eric

**Comment by spacecowboy** [comment]
This is a problem I have been looking at for a long time but not been able to solve. I'm afraid I no longer actively use the app myself so don't have time to maintain it. I'd be happy to accept any PRs and publish bug fixes though.

If you want to look deeper into this, the relevant file is this: https://github.com/spacecowboy/NotePad/blob/master/app/src/main/java/com/nononsenseapps/notepad/data/remote/gtasks/GoogleTaskSync.java

As you can see, there are a number of debug log points. Typically I have seen issues with conflict handling, as well as random request failures for certain tasks which terminate the entire sync.

**Comment by licaon-kter** [comment]
>  I'm afraid I no longer actively use the app myself 

What do you use (if you use such an app that is) ?

**Comment by spacecowboy** [comment]
For the odd note I still use the latest build. But I hardly take any notes. Any notes I do take are only valid for the day, like shopping lists or whatever. So I make no use of any synchronization.

**Comment by ericzolf** [comment]
So, I've forked the project yesterday, imported it in the latest Android Studio, compiled, installed and started it successfully on my smartphone. Then it failed syncing but I've now read your README, so I know that I need to and where to look for an API key... (stupid me)

Anyway, also interesting: just after I reinstalled the app from Google Play, the first sync failed, but it worked again after I had shut OFF and back ON the sync with Google. Perhaps a simpler workaround, I need to check if I can repeat this feat.

Another question: Studio bugs to me upgrade the project to the latest version of Gradle for instant run (or something like that). Would it be a valid Pull Request if I do it?
